authority/provisioner/acme.go

```diff
@@ -163,3 +168,21 @@ func (p *ACME) AuthorizeRevoke(ctx context.Context, token string) error {
 func (p *ACME) AuthorizeRenew(ctx context.Context, cert *x509.Certificate) error {
 	return p.ctl.AuthorizeRenew(ctx, cert)
 }
+
+// AuthorizeChallenge checks if the given challenge is enabled. By default
+// http-01, dns-01 and tls-alpn-01 are enabled, to disable any of them the
+// Challenge provisioner property should have at least one element.
+func (p *ACME) AuthorizeChallenge(ctx context.Context, challenge string) error {
+	enabledChallenges := []string{
+		"http-01", "dns-01", "tls-alpn-01",
+	}
+	if len(p.Challenges) > 0 {
+		enabledChallenges = p.Challenges
+	}
```

Comment:
Alternatively this could be done in the `Init()` once

Reply:
Unless you use a private variable, doing it on the init might change how the configuration is stored, it will be consistent, but I prefer blank defaults as they might change.